ocs_ci/ocs/resources/mcg.py

```diff
@@ -907,91 +913,57 @@ def exec_mcg_cmd(self, cmd, namespace=None, use_yes=False, **kwargs):
     )
     def retrieve_noobaa_cli_binary(self):
         """
-        Copy the NooBaa CLI binary from the operator pod
-        if it wasn't found locally, or if the hashes between
-        the two don't match.
+        Download the MCG CLI from quay based on OS type
 
         Raises:
-            NoobaaCliChecksumFailedException: If checksum doesn't match.
-            AssertionError: In the case CLI binary doesn't exist.
+            UnsupportedOSType: OS type is not supported.
 
         """
+        ocs_version = get_ocs_build_number()
+        pull_secret_path = os.path.join(constants.DATA_DIR, "pull-secret")
+        mcg_cli_image_location = os.path.join(
+            config.ENV_DATA["cluster_path"], f"mcg-cli-images-{config.RUN['run_id']}"
+        )
+        create_directory_path(mcg_cli_image_location)
 
```

Comment:
Inspired by a similar part in #6192, I think we should retrieve the pull-secret from the cluster to avoid adding an additional manual set-up step:

```suggestion
secret_obj = OCP(kind="secret", namespace=constants.OPENSHIFT_CONFIG_NAMESPACE)
pull_secret_obj = secret_obj.get(resource_name=constants.CLUSTER_REGISTRY_CONFIG_SECRET)
pull_secret = base64.b64decode(pull_secret_obj['data']['".dockerconfigjson"'])

with open(pull_secret_path, 'w') as pull_secret_file:
    pull_secret_file.write(pull_secret)
```